shared/pkg/job_factory/factory.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package job_factory // nolint:revive,stylecheck
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/kubeclarity/shared/pkg/job_manager"
+)
+
+var (
+	createJobFuncs map[string]CreateJobFunc // scanner name to CreateJobFunc
+	once           sync.Once
+)
+
+type CreateJobFunc func(conf job_manager.IsConfig, logger *logrus.Entry, resultChan chan job_manager.Result) job_manager.Job
+
+func RegisterCreateJobFunc(name string, createJobFunc CreateJobFunc) {
+	once.Do(func() {
+		createJobFuncs = make(map[string]CreateJobFunc)
+	})
+	if _, ok := createJobFuncs[name]; ok {
+		logrus.Fatalf("%q already registered", name)
+	}
+	createJobFuncs[name] = createJobFunc
+}
+
+func GetCreateJobFuncs() map[string]CreateJobFunc {
+	return createJobFuncs
+}
```

Comment:
See my other comment for more details but I think this should look something like:
```
type Factory struct {
    jobFuncs map[string]CreateJobFunc
}

func (f *Factory) Register(name string, createJobFunc CreateJobFunc) {
    if f.jobFuncs == nil {
        f.jobFuncs = map[string]CreateJobFunc{}
    }
    if _, ok := f.jobFuncs[name]; ok {
	logrus.Fatalf("%q already registered", name)
    } 
    f.jobFuncs[name] = createJobFunc
}

func (f *Factory) CreateJob(name string, conf job_manager.IsConfig, logger *logrus.Entry, resultChan chan job_manager.Result) (job_manager.Job, error) {
    createFunc, ok := f.jobFuncs[name]
    if !ok {
       return nil, fmt.Errorf("%v not a registered job", name)
    }
    return createFunc(conf, logger, resultChan)
} 
```